step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/IsolatedAutomationPackageRepository.java

```diff
@@ -83,8 +85,15 @@ public ArtefactInfo getArtefactInfo(Map<String, String> repositoryParameters) {
         }
 
         ArtefactInfo info = new ArtefactInfo();
-        info.setType("automationPackage");
-        info.setName(resource.getCustomField(AP_NAME_CUSTOM_FIELD, String.class));
+        boolean isWrapInTestSet = Boolean.parseBoolean(repositoryParameters.getOrDefault(ArtifactRepositoryConstants.PARAM_WRAP_PLANS_INTO_TEST_SET, "false"));
+        //Introduced for 28.1, previous execution do not have this field in repo parameters
+        String rootType = repositoryParameters.get(ArtifactRepositoryConstants.PARAM_ROOT_TYPE);
+        if (rootType != null) {
+            info.setType(rootType);
+        } else {
+            info.setType((isWrapInTestSet) ? TestSet.class.getSimpleName() : TestCase.class.getSimpleName());
+        }
+        info.setName((isWrapInTestSet) ? resource.getCustomField(AP_NAME_CUSTOM_FIELD, String.class) : repositoryParameters.getOrDefault(ArtifactRepositoryConstants.PARAM_INCLUDE_PLANS, apName));
```

Comment:
Not sure to understand why we use the concatenation of the selected plans as name here. What is the reason? It doesn't seem to be consistant with other repositories (azure, etc)